### ISSUE TITLE
Date#getUTCDate/getDate/toISOString issues in Opera 10.6

### ISSUE THREAD

**Issue by ljharb** [es-shims/es5-shim#240]
``` js
var d = new Date(-3509827334573292);
assert(d.getMonth() === 0);
assert(d.getUTCMonth() === 0);
```

Both of these fail in Opera 10.6.

**Comment by ljharb** [comment]
Added tests in f05e4236eec7fdefef5f82e731e544acdfed7919

**Comment by Xotic750** [comment]
Just been having a look at this. That's really quite a broken mess. Fixable by replacing all affected methods by the looks of things (quite a few, the 2 you mention, all the `toString` methods, and the milliseconds), but a real mess. Didn't they think to test `Date` before shipping?!!

**Comment by Xotic750** [comment]
I've downloaded and installed Opera 10.6 on my systems and started doing just that. I commented out all the tests, it crashed. I commented out all the code, it ran. I started putting code back in and this line (may not be the only one) caused it to crash when reinstated.
```
var hasToDateStringFormatBug = negativeTestDate.toDateString() !== 'Thu Jan 01 -109252';
```

**Comment by ljharb** [comment]
and it's not try/catch-able?

**Comment by Xotic750** [comment]
Haven't tried catch or anything yet, but `toUTCString` also crashes with the original test dates, but not the new date that I posted

**Comment by Xotic750** [comment]
Nope, try...catch does nothing, it just bums out saying core dump

**Comment by Xotic750** [comment]
I'll see if I can use the original test dates for everything except the ones using `toDateString` and `toUTCString`, but I guess there is little that be done about the core dump

**Comment by Xotic750** [comment]
Whoops, just noticed that I posted these comments here rather than in the PR, sorry!

**Comment by ljharb** [comment]
Updated list of Opera 10.6 failures:
 - #getUTCDate()
  - should return the right value for negative dates
   - Expected -32965446 to be 1, Date(Thu Jan 0-32965446 -109252 03:59:59 GMT-0800).
 - #getDate()
  - should return the right value for negative dates
   - Expected -32965446 to be 1.
 - #toISOString()
  - should return correct dates
   - Expected '-109252-08-78T11:59:59.708Z' to be '-109252-01-01T11:59:59.708Z'.

**Comment by Xotic750** [comment]
10.6
![bs_winxp_opera_10 6](https://cloud.githubusercontent.com/assets/216041/12643386/0db35c6a-c5bc-11e5-972d-c7bf32119528.jpg)
